src/utils.c

```diff
@@ -504,3 +504,30 @@ guint get_sectorsize(gint fd)
 
 	return sector_size;
 }
+
+goffset get_device_size(gint fd, GError **error)
+{
+	u_int64_t size = 0;
```

Comment:
Why not use glib type `guint64` here? An alternative would be using the original type from the `ioctl()` prototype which is `unsigned long`.